commit 78e3035e7b19
Author: Nathan Froyd <froydnj@mozilla.com>
Date:   Mon Oct 9 16:11:10 2017 -0400

    Bug 1406474 - Provide gfxUserFontSet::Entry with a move constructor. r=heycam, a=ritu
    
    This change makes moving Entry around more efficient, and also copies
    the mAllowedFontSets member as a ride-along bonus fix.
    
    --HG--
    extra : source : 2b181f6ba9d4c0e82d3abc947b31270114425b5c
---
 gfx/thebes/gfxUserFontSet.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git gfx/thebes/gfxUserFontSet.h gfx/thebes/gfxUserFontSet.h
index 1c84ceb128d5..b71219901ca9 100644
--- gfx/thebes/gfxUserFontSet.h
+++ gfx/thebes/gfxUserFontSet.h
@@ -403,11 +403,12 @@ public:
                   mPrivate(aKey->mPrivate)
             { }
 
-            Entry(const Entry& aOther)
-                : mURI(aOther.mURI),
-                  mPrincipal(aOther.mPrincipal),
-                  mFontEntry(aOther.mFontEntry),
-                  mPrivate(aOther.mPrivate)
+            Entry(Entry&& aOther)
+                : mAllowedFontSets(mozilla::Move(aOther.mAllowedFontSets)),
+                  mURI(mozilla::Move(aOther.mURI)),
+                  mPrincipal(mozilla::Move(aOther.mPrincipal)),
+                  mFontEntry(mozilla::Move(aOther.mFontEntry)),
+                  mPrivate(mozilla::Move(aOther.mPrivate))
             { }
 
             ~Entry() { }
